### DISCUSSION TITLE
Command timeout and retry mechanism

### DISCUSSION THREAD

**Discussion by Ricciolo** [Koenkk/zigbee2mqtt#22908] [General]
Hi,

I have a few questions and doubts regarding the command timeout behavior in ZigBee2MQTT when controlling devices such as lights, specifically TUYA devices.

1. **Timeout Duration**: I noticed that when I send a command to turn on a light and the device is unavailable, the command times out after 5 seconds, even though the log message states a timeout of 10 seconds. Could you clarify whether the actual timeout is 10 seconds or 5 seconds?

2. **Automatic Retry**: If a device is disconnected or partially disconnected for a moment, does the timeout mechanism include an automatic retry for sending the command? Is this retry handled by the ZigBee protocol itself, or is it managed by ZigBee2MQTT?

3. **Timeout Configuration**: Is it possible to adjust the timeout duration for specific commands (via mqtt message)? The current timeout significantly impacts scenarios where multiple lights need to be turned on or off. Since the commands are serialized and executed one after the other, if one light has connectivity issues, the whole process slows down considerably. Customizing the timeout for specific states set could help mitigate this issue.

I appreciate any insights you can provide on these questions.

Thank you!

**Comment by StefanLobbenmeier** [comment]
I have a somewhat related issue that thermostats every once in a while do not react to a temperature change. I play on adding automations in home assistant to detect this and resend the configured temperature regularly, something like this could be part of zigbee2mqtt though
